src/components/Pagination.tsx

```diff
@@ -1,78 +1,107 @@
-import { Dispatch, SetStateAction, useMemo } from "react";
+import { Dispatch } from "react";
 import { IconButton } from "src/components";
 import { Css, Palette } from "src/Css";
 import { SelectField } from "src/inputs";
 import { useTestIds } from "src/utils";
 
-export type PageSettings = {
-  page: number;
-  perPage: number;
-};
+/**
+ * Page settings, either a pageNumber+pageSize or offset+limit.
+ *
+ * This component is implemented in terms of "page number + page size",
+ * but our backend wants offset+limit, so we accept both and translate.
+ */
+export type PageSettings = PageNumberAndSize | OffsetAndLimit;
+export type PageNumberAndSize = { pageNumber: number; pageSize: number };
+export type OffsetAndLimit = { offset: number; limit: number };
+
+// Use OffsetAndLimit as our default b/c that's what backend filters expect
+export const defaultPage: OffsetAndLimit = { offset: 0, limit: 100 };
 
 interface PaginationProps {
-  label: string;
+  page: readonly [PageNumberAndSize, Dispatch<PageNumberAndSize>] | readonly [OffsetAndLimit, Dispatch<OffsetAndLimit>];
   totalCount: number;
-  settings: PageSettings;
-  setSettings: Dispatch<SetStateAction<PageSettings>>;
 }
 
 export function Pagination(props: PaginationProps) {
-  const { label, settings, totalCount, setSettings } = props;
-  const { perPage, page } = settings;
+  const { totalCount } = props;
+  const [page, setPage] = props.page;
+  const { pageSize, pageNumber } = toPageNumberSize(page);
+
+  const hasPrevPage = pageNumber > 1;
+  const hasNextPage = pageNumber < totalCount / pageSize;
+  // Create the `1 - 100 of 1000` or `0 of 0`
+  const first = pageSize * (pageNumber - 1) + (totalCount ? 1 : 0);
+  const last = Math.min(pageSize * pageNumber, totalCount);
+  // Don't both with `0 - 0 of 0`
+  const showLast = totalCount > 0;
 
-  const hasNextPage = useMemo(() => page < totalCount / perPage, [page, perPage, totalCount]);
+  // Convert the new page back to whatever format the caller wants
+  function set(newPage: PageNumberAndSize): void {
+    if ("pageNumber" in props.page[0]) {
+      setPage(newPage as any);
+    } else {
+      setPage(toLimitAndOffset(newPage) as any);
```

Comment:
**question:** Can we do something besides the `as any` here? Wondering if this was something that you added while it was a WIP, or we really just need to do this to make TS happy.

Reply:
I tried, but nope...not entirely sure why...really seems like it should worked.